pfsimulator/parflow_lib/overlandflow_eval_Kin.c

```diff
@@ -25,7 +25,7 @@
 
 #include "parflow.h"
 
-#if PARFLOW_ACC_BACKEND != PARFLOW_BACKEND_CUDA
+#if !defined(PARFLOW_HAVE_CUDA) && !defined(PARFLOW_HAVE_KOKKOS)
```

Comment:
I thought the pattern being established was to  check PARFLOW_ACC_BACKEND in non-backend specific code and the HAVE_CUDA etc was used only in backend specific files.

It looks like we have multiple defines for the same thing.  What is the difference?

Reply:
I decided to remove the whole PARFLOW_ACC_BACKEND thing completely, because it felt problematic and just use PARFLOW_HAVE_XXXXX everywhere. Just one definition now. It became less readable when not having CUDA or KOKKOS for example. Or with some other combinations.

Doing that also allows using the following mixed combination of configure args for development purposes (helps profiling the performance differences between CUDA and Kokkos, not intended for users): 
```-DPARFLOW_ACCELERATOR_BACKEND=cuda -DKokkos_ROOT=/path/to/Kokkos``` 

Basically, with this option, cudaMemset is used for initializations, and Kokkos uses some CUDA-specific template arguments, eg, instead of calling architecture-agnostic
```Kokkos::kokkos_malloc<Kokkos::HostSpace>(size)```
the CUDA-specific host memory allocation is used which allocates page-locked host memory for faster host-device data transfer:
```Kokkos::kokkos_malloc<Kokkos::CudaHostPinnedSpace>(size)```

Reply:
OK, I understand now.  Thanks.